nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/sensitive/aws/kms/AWSKMSSensitivePropertyProvider.java

```diff
@@ -0,0 +1,160 @@
+ */
+package org.apache.nifi.properties.sensitive.aws.kms;
+
+import com.amazonaws.services.kms.AWSKMS;
+import com.amazonaws.services.kms.AWSKMSClientBuilder;
+import com.amazonaws.services.kms.model.DecryptRequest;
+import com.amazonaws.services.kms.model.DecryptResult;
+import com.amazonaws.services.kms.model.EncryptRequest;
+import com.amazonaws.services.kms.model.EncryptResult;
+import java.nio.ByteBuffer;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.properties.sensitive.SensitivePropertyMetadata;
+import org.apache.nifi.properties.sensitive.SensitivePropertyProtectionException;
+import org.apache.nifi.properties.sensitive.SensitivePropertyProvider;
+import org.bouncycastle.util.encoders.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * This provider uses the AWS SDK to interact with the AWS KMS.  Values are encoded/decoded base64, using the
+ * standard encoders from bouncycastle.
+ */
+public class AWSKMSSensitivePropertyProvider implements SensitivePropertyProvider {
+    private static final Logger logger = LoggerFactory.getLogger(AWSKMSSensitivePropertyProvider.class);
+
+    private static final String IMPLEMENTATION_NAME = "AWS KMS Sensitive Property Provider";
+    protected static final String IMPLEMENTATION_KEY = "aws/kms/";
+
+    private AWSKMS client;
+    private final String key;
+
+    public AWSKMSSensitivePropertyProvider(String keyId) {
+        this.key = validateKey(keyId);
+        this.client = AWSKMSClientBuilder.standard().build();
+    }
+
+    private String validateKey(String keyId) {
+        if (keyId == null || StringUtils.isBlank(keyId)) {
+            throw new SensitivePropertyProtectionException("The key cannot be empty");
+        }
+        if (keyId.startsWith(IMPLEMENTATION_KEY)) {
+            keyId = keyId.substring(IMPLEMENTATION_KEY.length());
+        }
+        return keyId;
+    }
+
+    /**
+     * Returns the name of the underlying implementation.
+     *
+     * @return the name of this sensitive property provider
+     */
+    @Override
+    public String getName() {
+        return IMPLEMENTATION_NAME;
+
+    }
+
+    /**
+     * Returns the key used to identify the provider implementation in {@code nifi.properties}.
+     *
+     * @return the key to persist in the sibling property
+     */
+    @Override
+    public String getIdentifierKey() {
+        return IMPLEMENTATION_KEY + key; // getIdentifierKey() has to include the kms key id/alias/arn
+    }
+
+
+    /**
+     * Returns the encrypted cipher text.
+     *
+     * @param unprotectedValue the sensitive value
+     * @return the value to persist in the {@code nifi.properties} file
+     * @throws SensitivePropertyProtectionException if there is an exception encrypting the value
+     */
+    @Override
+    public String protect(String unprotectedValue) throws SensitivePropertyProtectionException {
+        if (unprotectedValue == null || unprotectedValue.trim().length() == 0) {
```

Comment:
Using `StringUtils.isBlank(unprotectedValue)` is beneficial because it encapsulates the trim and also checks if the content is all whitespace.

Reply:
Good catch, updated.